src/saturn_engine/utils/lru.py

```diff
@@ -0,0 +1,30 @@
+from collections import OrderedDict
+
+
+class LRUCounter(OrderedDict):
+    """Dict with a limited length, ejecting LRUs as needed."""
+
+    def __init__(self, *args, cache_len: int = 10, default_value: int = 0, **kwargs):
```

Comment:
Missing not sure about `default_value`.

I would suggest two options:

* Make it an `LRUCounter` and only expose `get(key)` and `add(key)`
* Make it an `LRUDefaultDict` and use a `default_factory` 

WDYT?

Reply:
I choose the second options : p